### PR TITLE
Remove `compute_expansion_coefficients` and store dual matrix

### PR DESCRIPTION
**Pull request by mscroggs**
Removed the `compute_expansion_coefficients` function, and move the computation it did inside the `FiniteElement` constructor.

Removed `_matM_new` from `FiniteElement` as is it unused.

Added `dual_matrix()` and `coefficient_matrix()` methods so these matrices can be accessed from Python

### CODE REVIEW

File: cpp/basix/finite-element.h [hunk 1]
```diff
@@ -773,6 +756,15 @@ class FiniteElement
...
                    const xtl::span<const T>& data, const int block_size) const;
 
+  /// Get the dual matrix.
+  ///
+  /// This is the matrix @f$BD^{T}@f$, as described in the documentation of the
+  /// `FiniteElement()` constructor.
+  /// @return The dual matrix
+  xt::xtensor<double, 2> dual_matrix() const;
+
+  xt::xtensor<double, 2> coefficient_matrix() const;
```
Comment (garth-wells) [inline cpp/basix/finite-element.h:766]: Doc string is missing.

File: cpp/basix/cell.cpp [hunk 2]
```diff
@@ -15,7 +15,7 @@ xt::xtensor<double, 2> cell::geometry(cell::type celltype)
   switch (celltype)
   {
   case cell::type::point:
-    return xt::xtensor<double, 2>({{0.0}});
+    return xt::xtensor<double, 2>({{}});
```
Comment (garth-wells) [inline cpp/basix/cell.cpp:18]: Is this change intended?

Comment (mscroggs) [reply cpp/basix/cell.cpp:18]: Maybe.

I think the shape of `cell::geometry` should be npoints by tdim, so in this case 1 by 0.

On the other hand, this isn't the change that fixed the space on points, so we could alternatively have the shape as npoints by gdim and declare that the gdim for a point is 1.

**Review by garth-wells — APPROVED** [review]
